### ISSUE TITLE
Improve confirmation dialogs validation

### ISSUE THREAD

**Issue by sunnyking** [virtualeconomy/v-wallet-gui#383]
With regard to #354 , the confirmation dialogs fail to detect the problem. Ningchen reported that the first confirmation dialog displayed mismatching info to the QR generated in the next step. The confirmation dialogs (1st hot wallet, 2nd cold wallet, 3rd hot wallet) needs to be reexamined for correctness.

**Comment by ncying** [comment]
I think @sunnyking means that in the last confirm dialog, the information like the sender, recipient and amount... should decode from the `dataObject` which send to the cold wallet not direct from the local variables. It should be the same if the whole procedure is correct (it should be right since the signature is verified before the confirm page). But if the data mismatch or local variables changed after signature validation, then `dataObject` which send to the cold wallet has a higher priority.

e.g.
```javascript
          <Confirm :address=showAddress()
                   :recipient=showRecipient()
                   :amount=inputAmount(amount)
                   :fee="fee"
                   :description="description"
                   :tx-type="'payment'">
          </Confirm>
```

**Comment by SheldonYS** [comment]
fixed by #385
